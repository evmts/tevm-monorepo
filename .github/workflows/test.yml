on:
  push:
    branches:
      - main
  pull_request:
  merge_group:

name: Test

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          submodules: recursive

      - name: "Setup"
        uses: ./.github/actions/setup

      - name: Run tests
        run: bun nx run-many --target=test:coverage
        env:
          NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
          RPC_URL_MAINNET: ${{ secrets.RPC_URL_MAINNET }}

      - name: 'Report @evmts/core Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/core"
          working-directory: "./core"
      - name: 'Report @evmts/ethers Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ethers"
          working-directory: "./ethers"
      - name: 'Report @evmts/bundler Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/bundler"
          working-directory: "./bundlers/bundler"
      - name: 'Report @evmts/bun-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/bun-plugin"
          working-directory: "./bundlers/bun"
      - name: 'Report @evmts/esbuild-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/esbuild-plugin"
          working-directory: "./bundlers/esbuild"
      - name: 'Report @evmts/ts-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ts-plugin"
          working-directory: "./ts-plugin"
      - name: 'Report @evmts/config Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/config"
          working-directory: "./config"
      - name: 'Report @evmts/esbuild Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/esbuild"
          working-directory: "./examples/esbuild"
  e2e:
    name: E2E
    timeout-minutes: 60
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          submodules: recursive

      - name: "Setup"
        uses: ./.github/actions/setup

      - name: Install Playwright Browsers
        run: npx playwright install --with-deps

      - name: Run tests
        run: echo "WIP" # bun nx e2e e2e
        env:
          NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
