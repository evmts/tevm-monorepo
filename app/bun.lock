{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "app",
      "dependencies": {
        "electrobun": "^0.0.18",
      },
      "devDependencies": {
        "@types/bun": "latest",
      },
      "peerDependencies": {
        "typescript": "^5",
      },
    },
  },
  "packages": {
    "@oneidentity/zstd-js": ["@oneidentity/zstd-js@1.0.3", "", { "dependencies": { "@types/emscripten": "^1.39.4" } }, "sha512-Jm6sawqxLzBrjC4sg2BeXToa33yPzUmq20CKsehKY2++D/gHb/oSwVjNgT+RH4vys+r8FynrgcNzGwhZWMLzfQ=="],

    "@types/bun": ["@types/bun@1.2.13", "", { "dependencies": { "bun-types": "1.2.13" } }, "sha512-u6vXep/i9VBxoJl3GjZsl/BFIsvML8DfVDO0RYLEwtSZSp981kEO1V5NwRcO1CPJ7AmvpbnDCiMKo3JvbDEjAg=="],

    "@types/emscripten": ["@types/emscripten@1.40.1", "", {}, "sha512-sr53lnYkQNhjHNN0oJDdUm5564biioI5DuOpycufDVK7D3y+GR3oUswe2rlwY1nPNyusHbrJ9WoTyIHl4/Bpwg=="],

    "@types/filesystem": ["@types/filesystem@0.0.36", "", { "dependencies": { "@types/filewriter": "*" } }, "sha512-vPDXOZuannb9FZdxgHnqSwAG/jvdGM8Wq+6N4D/d80z+D4HWH+bItqsZaVRQykAn6WEVeEkLm2oQigyHtgb0RA=="],

    "@types/filewriter": ["@types/filewriter@0.0.33", "", {}, "sha512-xFU8ZXTw4gd358lb2jw25nxY9QAgqn2+bKKjKOYfNCzN4DKCFetK7sPtrlpg66Ywe3vWY9FNxprZawAh9wfJ3g=="],

    "@types/har-format": ["@types/har-format@1.2.16", "", {}, "sha512-fluxdy7ryD3MV6h8pTfTYpy/xQzCFC7m89nOH9y94cNqJ1mDIDPut7MnRHI3F6qRmh/cT2fUjG1MLdCNb4hE9A=="],

    "@types/node": ["@types/node@22.15.18", "", { "dependencies": { "undici-types": "~6.21.0" } }, "sha512-v1DKRfUdyW+jJhZNEI1PYy29S2YRxMV5AOO/x/SjKmW0acCIOqmbj6Haf9eHAhsPmrhlHSxEhv/1WszcLWV4cg=="],

    "@types/webextension-polyfill": ["@types/webextension-polyfill@0.12.3", "", {}, "sha512-F58aDVSeN/MjUGazXo/cPsmR76EvqQhQ1v4x23hFjUX0cfAJYE+JBWwiOGW36/VJGGxoH74sVlRIF3z7SJCKyg=="],

    "browser-namespace": ["browser-namespace@1.4.0", "", { "dependencies": { "@types/filesystem": "*", "@types/har-format": "*", "@types/webextension-polyfill": "*" } }, "sha512-9b4yNTNs+8HVPssSq8RSZMRunf+G4cVQ2PMtOTn+uEVFOW5C0Uo+eGXuJ5LfxS1UDph5oAdWj92thPyxVhpqXg=="],

    "bun-types": ["bun-types@1.2.13", "", { "dependencies": { "@types/node": "*" } }, "sha512-rRjA1T6n7wto4gxhAO/ErZEtOXyEZEmnIHQfl0Dt1QQSB4QV0iP6BZ9/YB5fZaHFQ2dwHFrmPaRQ9GGMX01k9Q=="],

    "chownr": ["chownr@2.0.0", "", {}, "sha512-bIomtDF5KGpdogkLd9VspvFzk9KfpyyGlS8YFVZl7TGPBHL5snIOnxeshwVgPteQ9b4Eydl+pVbIyE1DcvCWgQ=="],

    "electrobun": ["electrobun@0.0.18", "", { "dependencies": { "@oneidentity/zstd-js": "^1.0.3", "rpc-anywhere": "1.5.0", "tar": "^6.2.1" }, "bin": { "electrobun": "dist/electrobun" } }, "sha512-RyMNGcAaHklicZlJToGfN3fVZGKHpxZv6o8S96TK9tHSY/SRze5bNPIGUnY9wr/BbWuQW5gGUGaVIWWqa5NSZQ=="],

    "fs-minipass": ["fs-minipass@2.1.0", "", { "dependencies": { "minipass": "^3.0.0" } }, "sha512-V/JgOLFCS+R6Vcq0slCuaeWEdNC3ouDlJMNIsacH2VtALiu9mV4LPrHc5cDl8k5aw6J8jwgWWpiTo5RYhmIzvg=="],

    "minipass": ["minipass@5.0.0", "", {}, "sha512-3FnjYuehv9k6ovOEbyOswadCDPX1piCfhV8ncmYtHOjuPwylVWsghTLo7rabjC3Rx5xD4HDx8Wm1xnMF7S5qFQ=="],

    "minizlib": ["minizlib@2.1.2", "", { "dependencies": { "minipass": "^3.0.0", "yallist": "^4.0.0" } }, "sha512-bAxsR8BVfj60DWXHE3u30oHzfl4G7khkSuPW+qvpd7jFRHm7dLxOjUk1EHACJ/hxLY8phGJ0YhYHZo7jil7Qdg=="],

    "mkdirp": ["mkdirp@1.0.4", "", { "bin": { "mkdirp": "bin/cmd.js" } }, "sha512-vVqVZQyf3WLx2Shd0qJ9xuvqgAyKPLAiqITEtqW0oIUjzo3PePDd6fW9iFz30ef7Ysp/oiWqbhszeGWW2T6Gzw=="],

    "rpc-anywhere": ["rpc-anywhere@1.5.0", "", { "dependencies": { "browser-namespace": "^1.4.0" } }, "sha512-ZYrB0foAM4oE7oBnUH3BL7LwtW9d6+RkzL/rFnjj8GCaFt5c81Rbw6oVl6u9AMsGONsKeJX0mL62TpbPXSO6og=="],

    "tar": ["tar@6.2.1", "", { "dependencies": { "chownr": "^2.0.0", "fs-minipass": "^2.0.0", "minipass": "^5.0.0", "minizlib": "^2.1.1", "mkdirp": "^1.0.3", "yallist": "^4.0.0" } }, "sha512-DZ4yORTwrbTj/7MZYq2w+/ZFdI6OZ/f9SFHR+71gIVUZhOQPHzVCLpvRnPgyaMpfWxxk/4ONva3GQSyNIKRv6A=="],

    "typescript": ["typescript@5.8.3", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-p1diW6TqL9L07nNxvRMM7hMMw4c5XOo/1ibL4aAIGmSAt9slTE1Xgw5KWuof2uTOvCg9BY7ZRi+GaF+7sfgPeQ=="],

    "undici-types": ["undici-types@6.21.0", "", {}, "sha512-iwDZqg0QAGrg9Rav5H4n0M64c3mkR59cJ6wQp+7C4nI0gsmExaedaYLNO44eT4AtBBwjbTiGPMlt2Md0T9H9JQ=="],

    "yallist": ["yallist@4.0.0", "", {}, "sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A=="],

    "fs-minipass/minipass": ["minipass@3.3.6", "", { "dependencies": { "yallist": "^4.0.0" } }, "sha512-DxiNidxSEK+tHG6zOIklvNOwm3hvCrbUrdtzY74U6HKTJxvIDfOUL5W5P2Ghd3DTkhhKPYGqeNUIh5qcM4YBfw=="],

    "minizlib/minipass": ["minipass@3.3.6", "", { "dependencies": { "yallist": "^4.0.0" } }, "sha512-DxiNidxSEK+tHG6zOIklvNOwm3hvCrbUrdtzY74U6HKTJxvIDfOUL5W5P2Ghd3DTkhhKPYGqeNUIh5qcM4YBfw=="],
  }
}
