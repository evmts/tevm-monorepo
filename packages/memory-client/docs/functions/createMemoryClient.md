**@tevm/memory-client** ∙ [README](../README.md) ∙ [API](../API.md)

***

[API](../API.md) > createMemoryClient

# Function: createMemoryClient()

> **createMemoryClient**(`options`?): `Promise`\<[`MemoryClient`](../type-aliases/MemoryClient.md)\>

A local EVM instance running in JavaScript. Similar to Anvil in your browser

## Parameters

▪ **options?**: [`CreateEVMOptions`](../type-aliases/CreateEVMOptions.md)= `{}`

## Returns

## Example

```ts
import { createMemoryClient } from "tevm"
import { createPublicClient, http } from "viem"
import { MyERC721 } from './MyERC721.sol'

const tevm = createMemoryClient({
	fork: {
	  url: "https://mainnet.optimism.io",
	},
})

const address = '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045',

await tevm.runContractCall(
  MyERC721.write.mint({
    caller: address,
  }),
)

const balance = await tevm.runContractCall(
 MyERC721.read.balanceOf({
 caller: address,
 }),
 )
 console.log(balance) // 1n
 ```

## Source

[packages/memory-client/src/createMemoryClient.js:59](https://github.com/evmts/tevm-monorepo/blob/main/packages/memory-client/src/createMemoryClient.js#L59)

***
Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
