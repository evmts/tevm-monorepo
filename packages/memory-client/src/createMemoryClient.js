import { createBaseClient } from '@tevm/base-client'
import {
	eip1993EventEmitter,
	ethActions,
	requestEip1193,
	tevmActions,
	tevmSend,
} from '@tevm/decorators'

/**
 * A local EVM instance running in JavaScript. Similar to Anvil in your browser
 * @param {import('@tevm/base-client').BaseClientOptions} [options]
 * @returns {import('./MemoryClient.js').MemoryClient}
 * @example
 * ```ts
 * import { createMemoryClient } from "tevm"
 * import { MyERC721 } from './MyERC721.sol'
 *
 * const tevm = createMemoryClient({
 * 	fork: {
 * 	  url: "https://mainnet.optimism.io",
 * 	},
 * })
 *
 * const address = '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045',

 * await tevm.runContractCall(
 *   MyERC721.write.mint({
 *     caller: address,
 *   }),
 * )
 *
 * const balance = await tevm.runContractCall(
 *  MyERC721.read.balanceOf({
 *  caller: address,
 *  }),
 *  )
 *  console.log(balance) // 1n
 *  ```
 */
export const createMemoryClient = (options) => {
	return createBaseClient(options)
		.extend(tevmSend())
		.extend(eip1993EventEmitter())
		.extend(requestEip1193())
		.extend(ethActions())
		.extend(tevmActions())
}
