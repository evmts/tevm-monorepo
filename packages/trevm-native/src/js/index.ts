import { createLogger } from '@tevm/logger';
import { InternalError } from '@tevm/errors';
import type { StateManager } from '@tevm/state';
import type { Common } from '@tevm/common';
import type { Chain } from '@tevm/blockchain';

// Import native module
import { init_trevm, TrevmWrapper } from '../../dist/index.node';

export interface TrevmEvmOptions {
  common: Common;
  stateManager: StateManager;
  blockchain: Chain;
  customPrecompiles?: any[];
  profiler?: boolean;
  allowUnlimitedContractSize?: boolean;
  loggingLevel?: string;
}

export interface TrevmRunCallOptions {
  caller?: string;
  to?: string;
  data?: string;
  value?: bigint;
  gasLimit?: bigint;
  skipBalance?: boolean;
}

export interface TrevmRunCallResult {
  returnValue: string;
  executionGasUsed: string;
  error?: string;
}

export interface TrevmEvm {
  runCall(opts: TrevmRunCallOptions): Promise<TrevmRunCallResult>;
}

/**
 * Creates a trevm-based EVM instance
 * @param options EVM options
 * @returns A trevm EVM instance
 */
export const createTrevm = async ({
  common,
  stateManager,
  blockchain,
  customPrecompiles,
  profiler,
  allowUnlimitedContractSize,
  loggingLevel,
}: TrevmEvmOptions): Promise<TrevmEvm> => {
  const logger = createLogger({
    name: '@tevm/trevm',
    level: loggingLevel ?? 'warn',
  });
  
  logger.debug({
    allowUnlimitedContractSize,
    profiler,
    customPrecompiles: customPrecompiles?.map(c => c.address.toString()),
  });

  try {
    // Initialize the native trevm module with callbacks to the StateManager
    const nativeTrevm = await init_trevm({
      chainId: common.chainId(),
      allowUnlimitedContractSize: allowUnlimitedContractSize ?? false,
      stateCallbacks: {
        getAccount: async (address: string) => {
          try {
            const account = await stateManager.getAccount(address);
            if (!account) return null;
            
            return {
              nonce: account.nonce,
              balance: account.balance,
              codeHash: account.codeHash ? `0x${Buffer.from(account.codeHash).toString('hex')}` : undefined,
              code: account.code ? `0x${Buffer.from(account.code).toString('hex')}` : undefined,
            };
          } catch (error) {
            logger.error(`Error in getAccount for ${address}:`, error);
            return null;
          }
        },
        putAccount: async (address: string, account: any) => {
          try {
            await stateManager.putAccount(address, {
              nonce: account.nonce,
              balance: account.balance,
              codeHash: account.codeHash ? Buffer.from(account.codeHash.slice(2), 'hex') : undefined,
              code: account.code ? Buffer.from(account.code.slice(2), 'hex') : undefined,
            });
            return true;
          } catch (error) {
            logger.error(`Error in putAccount for ${address}:`, error);
            return false;
          }
        },
        getStorage: async (address: string, key: string) => {
          try {
            const value = await stateManager.getContractStorage(address, key);
            return value ? `0x${Buffer.from(value).toString('hex')}` : '0x0';
          } catch (error) {
            logger.error(`Error in getStorage for ${address} key ${key}:`, error);
            return '0x0';
          }
        },
        putStorage: async (address: string, key: string, value: string) => {
          try {
            await stateManager.putContractStorage(
              address,
              key,
              Buffer.from(value.slice(2), 'hex')
            );
            return true;
          } catch (error) {
            logger.error(`Error in putStorage for ${address} key ${key}:`, error);
            return false;
          }
        },
        getCode: async (address: string) => {
          try {
            const code = await stateManager.getContractCode(address);
            return code ? `0x${Buffer.from(code).toString('hex')}` : '0x';
          } catch (error) {
            logger.error(`Error in getCode for ${address}:`, error);
            return '0x';
          }
        },
        putCode: async (address: string, code: string) => {
          try {
            await stateManager.putContractCode(
              address,
              Buffer.from(code.slice(2), 'hex')
            );
            return true;
          } catch (error) {
            logger.error(`Error in putCode for ${address}:`, error);
            return false;
          }
        },
        getBlockHash: async (blockNumber: number) => {
          try {
            const block = await blockchain.getBlock(BigInt(blockNumber));
            return block ? `0x${Buffer.from(block.header.hash()).toString('hex')}` : null;
          } catch (error) {
            logger.error(`Error in getBlockHash for block ${blockNumber}:`, error);
            return null;
          }
        }
      }
    });

    // Return a JavaScript wrapper around the native trevm that matches the Evm interface
    return {
      async runCall(opts: TrevmRunCallOptions): Promise<TrevmRunCallResult> {
        try {
          return await nativeTrevm.run_call({
            caller: opts.caller,
            to: opts.to,
            data: opts.data,
            value: opts.value,
            gas_limit: opts.gasLimit,
            skip_balance: opts.skipBalance,
          });
        } catch (error) {
          logger.error('Error in runCall:', error);
          throw new InternalError(`Trevm execution error: ${error}`);
        }
      }
    };
  } catch (error) {
    logger.error('Error initializing trevm:', error);
    throw new InternalError(`Failed to initialize trevm: ${error}`);
  }
};