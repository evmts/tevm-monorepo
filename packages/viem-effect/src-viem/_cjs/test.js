'use strict'
Object.defineProperty(exports, '__esModule', { value: true })
exports.stopImpersonatingAccount =
	exports.setRpcUrl =
	exports.snapshot =
	exports.setStorageAt =
	exports.setNonce =
	exports.setNextBlockTimestamp =
	exports.setNextBlockBaseFeePerGas =
	exports.setMinGasPrice =
	exports.setLoggingEnabled =
	exports.setIntervalMining =
	exports.setCoinbase =
	exports.setCode =
	exports.setBlockTimestampInterval =
	exports.setBlockGasLimit =
	exports.setBalance =
	exports.setAutomine =
	exports.sendUnsignedTransaction =
	exports.revert =
	exports.reset =
	exports.removeBlockTimestampInterval =
	exports.mine =
	exports.inspectTxpool =
	exports.increaseTime =
	exports.impersonateAccount =
	exports.getTxpoolStatus =
	exports.getTxpoolContent =
	exports.getAutomine =
	exports.dropTransaction =
		void 0
const dropTransaction_js_1 = require('./actions/test/dropTransaction.js')
Object.defineProperty(exports, 'dropTransaction', {
	enumerable: true,
	get: function () {
		return dropTransaction_js_1.dropTransaction
	},
})
const getAutomine_js_1 = require('./actions/test/getAutomine.js')
Object.defineProperty(exports, 'getAutomine', {
	enumerable: true,
	get: function () {
		return getAutomine_js_1.getAutomine
	},
})
const getTxpoolContent_js_1 = require('./actions/test/getTxpoolContent.js')
Object.defineProperty(exports, 'getTxpoolContent', {
	enumerable: true,
	get: function () {
		return getTxpoolContent_js_1.getTxpoolContent
	},
})
const getTxpoolStatus_js_1 = require('./actions/test/getTxpoolStatus.js')
Object.defineProperty(exports, 'getTxpoolStatus', {
	enumerable: true,
	get: function () {
		return getTxpoolStatus_js_1.getTxpoolStatus
	},
})
const impersonateAccount_js_1 = require('./actions/test/impersonateAccount.js')
Object.defineProperty(exports, 'impersonateAccount', {
	enumerable: true,
	get: function () {
		return impersonateAccount_js_1.impersonateAccount
	},
})
const increaseTime_js_1 = require('./actions/test/increaseTime.js')
Object.defineProperty(exports, 'increaseTime', {
	enumerable: true,
	get: function () {
		return increaseTime_js_1.increaseTime
	},
})
const inspectTxpool_js_1 = require('./actions/test/inspectTxpool.js')
Object.defineProperty(exports, 'inspectTxpool', {
	enumerable: true,
	get: function () {
		return inspectTxpool_js_1.inspectTxpool
	},
})
const mine_js_1 = require('./actions/test/mine.js')
Object.defineProperty(exports, 'mine', {
	enumerable: true,
	get: function () {
		return mine_js_1.mine
	},
})
const removeBlockTimestampInterval_js_1 = require('./actions/test/removeBlockTimestampInterval.js')
Object.defineProperty(exports, 'removeBlockTimestampInterval', {
	enumerable: true,
	get: function () {
		return removeBlockTimestampInterval_js_1.removeBlockTimestampInterval
	},
})
const reset_js_1 = require('./actions/test/reset.js')
Object.defineProperty(exports, 'reset', {
	enumerable: true,
	get: function () {
		return reset_js_1.reset
	},
})
const revert_js_1 = require('./actions/test/revert.js')
Object.defineProperty(exports, 'revert', {
	enumerable: true,
	get: function () {
		return revert_js_1.revert
	},
})
const sendUnsignedTransaction_js_1 = require('./actions/test/sendUnsignedTransaction.js')
Object.defineProperty(exports, 'sendUnsignedTransaction', {
	enumerable: true,
	get: function () {
		return sendUnsignedTransaction_js_1.sendUnsignedTransaction
	},
})
const setAutomine_js_1 = require('./actions/test/setAutomine.js')
Object.defineProperty(exports, 'setAutomine', {
	enumerable: true,
	get: function () {
		return setAutomine_js_1.setAutomine
	},
})
const setBalance_js_1 = require('./actions/test/setBalance.js')
Object.defineProperty(exports, 'setBalance', {
	enumerable: true,
	get: function () {
		return setBalance_js_1.setBalance
	},
})
const setBlockGasLimit_js_1 = require('./actions/test/setBlockGasLimit.js')
Object.defineProperty(exports, 'setBlockGasLimit', {
	enumerable: true,
	get: function () {
		return setBlockGasLimit_js_1.setBlockGasLimit
	},
})
const setBlockTimestampInterval_js_1 = require('./actions/test/setBlockTimestampInterval.js')
Object.defineProperty(exports, 'setBlockTimestampInterval', {
	enumerable: true,
	get: function () {
		return setBlockTimestampInterval_js_1.setBlockTimestampInterval
	},
})
const setCode_js_1 = require('./actions/test/setCode.js')
Object.defineProperty(exports, 'setCode', {
	enumerable: true,
	get: function () {
		return setCode_js_1.setCode
	},
})
const setCoinbase_js_1 = require('./actions/test/setCoinbase.js')
Object.defineProperty(exports, 'setCoinbase', {
	enumerable: true,
	get: function () {
		return setCoinbase_js_1.setCoinbase
	},
})
const setIntervalMining_js_1 = require('./actions/test/setIntervalMining.js')
Object.defineProperty(exports, 'setIntervalMining', {
	enumerable: true,
	get: function () {
		return setIntervalMining_js_1.setIntervalMining
	},
})
const setLoggingEnabled_js_1 = require('./actions/test/setLoggingEnabled.js')
Object.defineProperty(exports, 'setLoggingEnabled', {
	enumerable: true,
	get: function () {
		return setLoggingEnabled_js_1.setLoggingEnabled
	},
})
const setMinGasPrice_js_1 = require('./actions/test/setMinGasPrice.js')
Object.defineProperty(exports, 'setMinGasPrice', {
	enumerable: true,
	get: function () {
		return setMinGasPrice_js_1.setMinGasPrice
	},
})
const setNextBlockBaseFeePerGas_js_1 = require('./actions/test/setNextBlockBaseFeePerGas.js')
Object.defineProperty(exports, 'setNextBlockBaseFeePerGas', {
	enumerable: true,
	get: function () {
		return setNextBlockBaseFeePerGas_js_1.setNextBlockBaseFeePerGas
	},
})
const setNextBlockTimestamp_js_1 = require('./actions/test/setNextBlockTimestamp.js')
Object.defineProperty(exports, 'setNextBlockTimestamp', {
	enumerable: true,
	get: function () {
		return setNextBlockTimestamp_js_1.setNextBlockTimestamp
	},
})
const setNonce_js_1 = require('./actions/test/setNonce.js')
Object.defineProperty(exports, 'setNonce', {
	enumerable: true,
	get: function () {
		return setNonce_js_1.setNonce
	},
})
const setStorageAt_js_1 = require('./actions/test/setStorageAt.js')
Object.defineProperty(exports, 'setStorageAt', {
	enumerable: true,
	get: function () {
		return setStorageAt_js_1.setStorageAt
	},
})
const snapshot_js_1 = require('./actions/test/snapshot.js')
Object.defineProperty(exports, 'snapshot', {
	enumerable: true,
	get: function () {
		return snapshot_js_1.snapshot
	},
})
const setRpcUrl_js_1 = require('./actions/test/setRpcUrl.js')
Object.defineProperty(exports, 'setRpcUrl', {
	enumerable: true,
	get: function () {
		return setRpcUrl_js_1.setRpcUrl
	},
})
const stopImpersonatingAccount_js_1 = require('./actions/test/stopImpersonatingAccount.js')
Object.defineProperty(exports, 'stopImpersonatingAccount', {
	enumerable: true,
	get: function () {
		return stopImpersonatingAccount_js_1.stopImpersonatingAccount
	},
})
//# sourceMappingURL=test.js.map
