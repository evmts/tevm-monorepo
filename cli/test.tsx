import React from 'react';
import chalk from 'chalk';
import test from 'ava';
import {render} from 'ink-testing-library';
import Index from './source/commands/index.js';

test('greet user', t => {
	const {lastFrame} = render(<Index options={{name: 'Jane'}} />);

	t.is(lastFrame(), `Hello, ${chalk.green('Jane')}`);
});
