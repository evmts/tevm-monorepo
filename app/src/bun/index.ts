import { BrowserWindow } from "electrobun/bun";

const win = new BrowserWindow({
  title: "Hello Electrobun",
  url: "views://ui/index.html",
});
