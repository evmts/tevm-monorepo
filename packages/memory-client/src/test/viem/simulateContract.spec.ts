import { beforeEach, describe, expect, it } from 'bun:test'
import { SimpleContract } from '@tevm/test-utils'
import type { MemoryClient } from '../../MemoryClient.js'
import { createMemoryClient } from '../../createMemoryClient.js'

let mc: MemoryClient<any, any>
let c = {
	simpleContract: SimpleContract.withAddress(`0x${'00'.repeat(20)}`),
}

beforeEach(async () => {
	mc = createMemoryClient()
	const deployResult = await mc.tevmDeploy({
		bytecode: SimpleContract.bytecode,
		abi: SimpleContract.abi,
		args: [420n],
	})
	if (!deployResult.createdAddress) {
		throw new Error('contract never deployed')
	}
	c = {
		simpleContract: SimpleContract.withAddress(deployResult.createdAddress),
	}
	if (!deployResult.txHash) {
		throw new Error('txHash not found')
	}
	await mc.tevmMine()
})

describe('simulateContract', () => {
	it('should work', async () => {
		expect(await mc.simulateContract(c.simpleContract.write.set(99999n))).toMatchSnapshot()
	})
})
