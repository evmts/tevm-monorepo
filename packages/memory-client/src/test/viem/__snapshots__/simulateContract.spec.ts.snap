// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`simulateContract > should work 1`] = `
{
  "request": {
    "abi": [
      {
        "inputs": [
          {
            "name": "newValue",
            "type": "uint256",
          },
        ],
        "name": "set",
        "outputs": [],
        "stateMutability": "nonpayable",
        "type": "function",
      },
    ],
    "account": undefined,
    "address": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
    "args": [
      99999n,
    ],
    "dataSuffix": undefined,
    "functionName": "set",
    "humanReadableAbi": [
      "function set(uint256 newValue)",
    ],
    "to": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
  },
  "result": undefined,
}
`;
